provisioning/hare_setup.py

```diff
@@ -0,0 +1,303 @@
+                exit(-1)
+
+            cluster_status = check_cluster_status()
+
+            while is_cluster_running():
+                shutdown_cluster()
+
+            if cluster_status:
+                logging.error('Cluster status reports failure')
+                exit(-1)
+
+        except Exception as error:
+            logging.error('Error while checking cluster status (%s)',
+                          error)
+            exit(-1)
+
+class SupportBundle(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        try:
+            # Default target directory is /tmp/hare
+            if os.system('hctl reportbug') != 0:
+                logging.error('Failed to generate support bundle')
+                exit(-1)
+        except Exception as error:
+            logging.error('Error while generating support bundle (%s)',
+                          error)
+            exit(-1)
+
+class Cleanup(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        try:
+            exit(0)
+        except Exception as error:
+            logging.error('Failed to perform cleanup (%s)',
+                          error)
+            exit(-1)
+
+def checkRpm(rpm_name):
+    rpm_list = subprocess.Popen(["rpm", "-qa"],
+                                stdin=subprocess.PIPE,
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE,
+                                env={},
+                                encoding='utf8')
+    rpm_search = subprocess.Popen(["grep", "-q", rpm_name],
+                                  stdin=rpm_list.stdout,
+                                  stdout=subprocess.PIPE,
+                                  stderr=subprocess.PIPE,
+                                  env={},
+                                  encoding='utf8')
+    out, err = rpm_search.communicate()
+    logging.debug("Output: {}".format(out))
+    logging.debug("stderr: {}".format(err))
+
+    return rpm_search.returncode
+
+
+def is_cluster_running() -> bool:
+    return os.system('hctl status >/dev/null') == 0
+
+
+def bootstrap_cluster():
+    return os.system('hctl bootstrap --mkfs /var/lib/hare/cluster.yaml')
+
+
+def shutdown_cluster():
+    return os.system('hctl shutdown')
+
+
+def list2dict(nodes_data_hctl: list) -> dict:
+    node_info_dict = {}
+    for node in nodes_data_hctl:
+        node_svc_info = {}
+        for service in node['svcs']:
+            if not service['name'] in node_svc_info.keys(): 
+                node_svc_info[service['name']] = []
+            if (service['status'] == 'started'):
+                node_svc_info[service['name']].append(service['status'])
+        node_info_dict[node['name']] = node_svc_info
+
+    return node_info_dict
+
+
+def check_cluster_status():
+    cluster_desc = None
+    with open("/var/lib/hare/cluster.yaml", 'r') as stream:
```

Comment:
I don't think it is a good idea to hardcode the cluster.yaml path. I think it would be good to accept it as an argument, either via conf-store or an explicit argument to hare-setup.py. But let's postpone this for now and we improve it later.